src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -0,0 +1,115 @@
+<!--
```

Comment:
Not sure bout this, but i think this file should be added the the ILLink nuget package as a Sdk.targets file. I see other SDKs do that too. @nguerrera What are your thoughts here?

Reply:
Example: see Microsoft.NET.Sdk.Razor, or Microsoft.NET.Sdk.Web

Reply:
i prefer to have a separate package. Although in this PR, ILTransform is part of Publish, that means it is not optional, does it have to?

Reply:
I'd like it to be in the SDK if possible. With the out-of-band linker package, we shipped our own targets that we had trouble keeping in sync with the SDK (see https://github.com/dotnet/sdk/issues/2583 for a taste of the issues).
- One advantage that you already pointed out is that we don't have to use BeforeTargets/AfterTargets as an extension point.
- There's also the dependency in `_ComputeUseBuildDependencyFile`: https://github.com/dotnet/sdk/pull/3125/files/3f83d196a8476f3889e2191e00c465e0c113b4a6#diff-7dc833d02f198e5a9f0ab88e229db18aR781.
- Finally, the use of `ResolvedFileToPublish` as an input is something that I hope we can replace with a more well-defined list of runtime assemblies (see https://github.com/dotnet/sdk/issues/3109), which is another case where we have a pretty tight coupling with the SDK.

Reply:
The mosnt important thing is that we have tests for the logic in these targets in the SDK repo. As long as we do that, it could live in either place but I think here makes sense. Curious what you think @nguerrera.

Reply:
Given that there is mutation of state, I would prefer to keep this here so we can have a pure contract with the linker.